full-service/src/service/sync.rs

```diff
@@ -121,13 +121,14 @@ pub fn sync_all_accounts(
     let accounts: Vec<Account> =
         { Account::list_all(conn, None, None).expect("Failed getting accounts from database") };
 
-    for account in accounts {
-        // If there are no new blocks for this account, don't do anything.
-        if account.next_block_index as u64 > num_blocks - 1 {
-            continue;
+    if num_blocks != 0 {
+        for account in accounts {
+            // If there are no new blocks for this account, don't do anything.
+            if account.next_block_index as u64 > num_blocks - 1 {
+                continue;
+            }
+            sync_account_next_chunk(ledger_db, conn, &account.id, logger)?;
         }
```

Comment:
```suggestion
    if num_blocks == 0 {
        Ok(())
    }
    for account in accounts {
        // If there are no new blocks for this account, don't do anything.
        if account.next_block_index as u64 > num_blocks - 1 {
            continue;
        }
        sync_account_next_chunk(ledger_db, conn, &account.id, logger)?;
    }
```

Reply:
Generally I think we want to try and avoid unnecessary nesting of code, so instead of wrapping the loop inside an if statement, we could instead invert the conditional and just return, otherwise continue with the execution of the program. This becomes exceedingly more and more important as conditions grow and become more complex.

Reply:
Also, not sure if that suggestion is commit worthy, so I would advise just implementing it yourself